src/lib/connectTraceToPlot.js

```diff
@@ -31,7 +31,7 @@ export default function connectTraceToPlot(WrappedComponent) {
 
       let fullTrace = {};
       for (let i = 0; i < fullData.length; i++) {
-        if (trace.uid === fullData[i]._fullInput._input.uid) {
+        if (traceIndexes[0] === fullData[i]._fullInput.index) {
```

Comment:
There's a traceIndexes array for each trace, presumably to identify traces that you get when you split or something like that? So seems like `traceIndex[0]` always points to THE trace. Or am I misunderstanding something?

Reply:
traceIndexes can be one single trace or an array of traces.
It will be an array when we're dealing with grouped traces, like in the accordions that allow you to style multiple traces at once.

Reply:
💃